glue/flagging_script_glue/flagging.py

```diff
@@ -767,114 +767,125 @@ def z_normalize_groupby(s: pd.Series):
-            (df["sv_short_owner"] == "Short-term owner")
-            & (df["sv_pricing"].str.contains("Low")),
-            (df["sv_name_match"] != "No match")
-            & (df["sv_pricing"].str.contains("Low")),
-            (
-                df[["sv_buyer_category", "sv_seller_category"]]
-                .eq("legal_entity")
-                .any(axis=1)
-            )
-            & (df["sv_pricing"].str.contains("Low")),
-            (df["sv_anomaly"] == "Outlier") & (df["sv_pricing"].str.contains("Low")),
-            (df["sv_pricing"].str.contains("Low price swing")),
-            (df["sv_pricing"].str.contains("Low"))
-            & (df["sv_which_price"] == "(raw & sqft)"),
-            (df["sv_pricing"].str.contains("Low")) & (df["sv_which_price"] == "(raw)"),
             (df["sv_pricing"].str.contains("Low")) & (df["sv_which_price"] == "(sqft)"),
         ]
 
-        labels = [
-            "Home flip sale (high)",
-            "Family sale (high)",
-            "Non-person sale (high)",
-            "Anomaly (high)",
-            "High price swing",
-            "High price (raw & sqft)",
-            "High price (raw)",
+        # Define labels for price-based reasons
+        price_labels = [
+            "High price",
+            "Low price",
             "High price (sqft)",
-            "Home flip sale (low)",
-            "Family sale (low)",
-            "Non-person sale (low)",
-            "Anomaly (low)",
-            "Low price swing",
-            "Low price (raw & sqft)",
-            "Low price (raw)",
             "Low price (sqft)",
         ]
 
-    df["sv_outlier_type"] = np.select(conditions, labels, default="Not outlier")
+    # Combined conditions and labels for reason1 with fallback to character conditions if no price condition is met
+    combined_conditions = price_conditions + char_conditions
+    combined_labels = price_labels + char_labels
+
+    df["sv_outlier_reason1"] = np.select(
+        combined_conditions, combined_labels, default="Not outlier"
+    )
+
+    # Adjust for reason2 based on remaining price conditions and char conditions not used in reason1
+    remaining_conditions = [
+        (cond & (df["sv_outlier_reason1"] != label))
+        for cond, label in zip(combined_conditions, combined_labels)
+    ]
+    df["sv_outlier_reason2"] = np.select(
+        remaining_conditions, combined_labels, default="Not outlier"
+    )
+
+    def find_next_unused_char(row, df):
+        used_labels = {row["sv_outlier_reason1"], row["sv_outlier_reason2"]}
+        for cond, label in zip(char_conditions, char_labels):
+            # Evaluate the condition directly for the current row
+            if label not in used_labels and cond[row.name]:
+                return label
+        return "Not outlier"
+
+    # Now we apply this function row-wise correctly by also passing the DataFrame reference:
+    df["sv_outlier_reason3"] = df.apply(find_next_unused_char, df=df, axis=1)
```

Comment:
Here we look to select a price outlier first (price is picked first by order of the `combined_conditions` and `combined_labels`) and if there are no price outliers, we assign a characteristic reason to populate `sv_outlier_reason1`. If this is the case, the observation won't be labeled as `1` for `sv_is_outlier` since we need a price outlier match in order to classify it as an outlier. However, the characteristic reason will remain for the `sv_outlier_reason1` column. 

Then, we take the next label that hasn't been already placed in `sv_outlier_reason1` and place it in `sv_outlier_reason2`. This could be another price label (raw and then sqft) or a characteristic option.  

For `sv_outlier_reason3` we look to see if there is another characteristic condition that has been satisfied.